### PR TITLE
Replace value for BLOCKING_ENABLED, add if it does not already exist

### PR DESCRIPTION
**Pull request by PromoFaux**
**By submitting this pull request, I confirm the following:** 

- [x] I have read and understood the [contributors guide](https://github.com/pi-hole/pi-hole/blob/master/CONTRIBUTING.md), as well as this entire template.
- [x] I have made only one major change in my proposed changes.
- [ ] I have commented my proposed changes within the code.
- [x] I have tested my proposed changes, and have included unit tests where possible.
- [x] I am willing to help maintain this change if there are issues with it later.
- [x] I give this submission freely and claim no ownership.
- [x] It is compatible with the [EUPL 1.2 license](https://opensource.org/licenses/EUPL-1.1)
- [x] I have squashed any insignificant commits. ([`git rebase`](http://gitready.com/advanced/2009/02/10/squashing-commits-with-rebase.html))

---
**What does this PR aim to accomplish?:**

Addresses [this comment](https://github.com/pi-hole/pi-hole/pull/4547#discussion_r783970521) by @MichaIng  by ensuring that `BLOCKING_ENABLED=[value]` will be created if it does not already exist in `setupVars.conf` for whatever reason. This would be the case on new install.

It also takes into account the initial reason for the change (introduced here https://github.com/pi-hole/pi-hole/pull/4555) which was that the order of `setupVars.conf` changed every time pihole was enabled/disabled (I am not 100% on why that matters, but maybe there is something I am unaware of that it breaks)

For those interested in the science : https://superuser.com/a/976712

### CODE REVIEW

File: pihole [hunk 1]
```diff
@@ -234,8 +234,8 @@ Time:
     fi
     echo -e "  ${INFO} Enabling blocking"
     local str="Pi-hole Enabled"
-
-    sed -i "s/^BLOCKING_ENABLED=false/BLOCKING_ENABLED=true/" "${setupVars}"
+
```
Comment (MichaIng) [inline pihole:237]: Obsolete spaces. Nice `sed` magic btw 👍🙂!

**Comment by MichaIng** [comment]
Could be also added here btw: https://github.com/pi-hole/pi-hole/blob/c208032/automated%20install/basic-install.sh#L1731-L1751
```
BLOCKING_ENABLED=${BLOCKING_ENABLED:-true}
```

**Comment by DL6ER** [comment]
So we know this `sed` magic works for all versions we might be seeing on user systems?

> I am not 100% on why that matters, but maybe there is something I am unaware of that it breaks

Probably they have comments in the file and want them to stay next to the lines always

File: pihole (hunk 1 above)
Comment (PromoFaux) [reply pihole]: All credit to the user on Superuser - I barely understand it :) 

I was going to try and functionise it, as it's used more than once, but... I don't care enough 😂

**Comment by PromoFaux** [comment]
> So we know this `sed` magic works for all versions we might be seeing on user systems?

😖 I can try and install on more systems. I only tried Ubuntu 20.04 and Debian 11 so far. 

There's always a [2-liner](https://superuser.com/a/1050172) that is probably almost guaranteed to work (and reads a bit easier):
```bash
grep -q '^FOOBAR=' somefile || echo 'FOOBAR=VALUE' >> somefile
sed -i 's/FOOBAR=.*$/FOOBAR=VALUE/' somefile 
```

**Comment by PromoFaux** [comment]
> Could be also added here btw

I wonder if there is any reason why it wasn't in the first place. Maybe it was once upon a time, or maybe it was just missed. But also adding it there would solve the problem for fresh installs without changing the `sed` commands at all (although then we would lose the guarding that replaces said values if they are not found)

**Comment by MichaIng** [comment]
Ah ignore my comment on the other PR about possible invalid values, since the `sed` replaces the present value of course.

But yes, to have something understandable for non-`sed`-magicians, the two liner is probably better, or even:
```sh
if grep -q '^FOOBAR=' somefile; then
    sed -i '/^FOOBAR=/c\FOOBAR=VALUE/' somefile 
else
    echo 'FOOBAR=VALUE' >> somefile
fi
```
to avoid doubled file writes at the cost of more lines 😉. The `sed` here btw uses the line replacement command `c` on the matching line, basically the same but a bit more explicit.

**Comment by PromoFaux** [comment]
This. I'm going to functionise this (I'll add you as co-author!)

**Comment by MichaIng** [comment]
Looks great now, good idea to move this into a function, can be used like that in other places 👍. To further reduce unnecessary writes when the exact value is present already:
```sh
grep -q "^${key}=${value}$" "${setupVars}" || sed -i "/^${key}=/c\\${key}=${value}" "${setupVars}"
```
But adds an additional command call in most cases (expecting values to be present and different usually), so probably not really an enhancement.

**Comment by PromoFaux** [comment]
I've now tested this function in all supported OSes (at least, in the official docker containers)

e.g  I have a file called `test.sh` in `/home/adam/playground/test/` and I mounted it into a container like so then ran it and observed the output

`docker run -it --rm -v '/home/adam/playground/test/:/test' fedora:33`

test.sh
```bash
#!/usr/bin/env bash

setupVars=/test/testfile

funAddOrEditKeyVal() {
  local key="${1}"
  local value="${2}"
  if grep -q "^${key}=" "${setupVars}"; then
    sed -i "/^${key}=/c\\${key}=${value}" "${setupVars}"
  else
    echo "${key}=${value}" >> "${setupVars}" 
  fi
}

touch $setupVars

funAddOrEditKeyVal KEY_ONE value1
funAddOrEditKeyVal KEY_TWO value2
funAddOrEditKeyVal KEY_THREE value3
funAddOrEditKeyVal KEY_ONE value4

chown 1000:1000 $setupVars
```

**Review by yubiuser — APPROVED** [review]

**Review by MichaIng — APPROVED** [review]

**Review by DL6ER — APPROVED** [review]

**Pull request merged**